### PR TITLE
fix: node selector support for node_exporter and fluent-bit

### PR DESCRIPTION
**Pull request by jdongo-aneo**
# Motivation
The node selectors are not applied to fluent-bit pods and node_exporters even if they are correctly defined.

# Description

Adding the defined node selectors to the daemonset definition

# Testing

N/A

# Impact
Pods will be created and scheduled only on the nodes satisfying the node selectors

# Additional Information

# Checklist

- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] I have thoroughly tested my modifications and added tests when necessary.
- [ ] Tests pass locally and in the CI.
- [ ] I have assessed the performance impact of my modifications.

### CODE REVIEW

**Review by tschneider-aneo — DISMISSED** [review]
Changes are functionnal but could be a bit cleaner IMO.

BTW, we could take advantage of this PR to clean how tolerations are implemented, for example : 

``` HCL
"tolerations" = var.node_selector ! {} ? [
  for key, value in var.node_selector : {
    "key"         = key
    "operator"    = "Equal" 
    "value"       = value
    "effect"      = "NoSchedule"
  }
] : []
```

File: monitoring/onpremise/exporters/node-exporter/locals.tf [hunk 1]
```diff
@@ -1,4 +1,5 @@
 locals {
-  node_selector_keys   = keys(var.node_selector)
-  node_selector_values = values(var.node_selector)
+  node_selector_keys          = keys(var.node_selector)
+  node_selector_values        = values(var.node_selector)
```
Comment (lemaitre-aneo) [inline monitoring/onpremise/exporters/node-exporter/locals.tf:3]: You can actually remove those and use the object directly in a for_each

File: monitoring/onpremise/exporters/node-exporter/main.tf [hunk 2]
```diff
@@ -32,17 +32,13 @@ resource "kubernetes_daemonset" "node_exporter" {
...
       spec {
+        node_selector = var.node_selector
         dynamic "toleration" {
-          for_each = (var.node_selector != {} ? [
-            for index in range(0, length(local.node_selector_keys)) : {
-              key   = local.node_selector_keys[index]
-              value = local.node_selector_values[index]
-            }
-          ] : [])
+          for_each = var.node_selector != {} ? var.node_selector : {}
```
Comment (lemaitre-aneo) [inline monitoring/onpremise/exporters/node-exporter/main.tf:37]:
```suggestion
          for_each = var.node_selector
```

File: monitoring/onpremise/fluent-bit/main.tf [hunk 3]
```diff
@@ -24,17 +24,13 @@ resource "kubernetes_daemonset" "fluent_bit" {
...
       spec {
+        node_selector = var.node_selector
         dynamic "toleration" {
-          for_each = (var.node_selector != {} ? [
-            for index in range(0, length(local.fluent_bit_node_selector_keys)) : {
-              key   = local.fluent_bit_node_selector_keys[index]
-              value = local.fluent_bit_node_selector_values[index]
-            }
-          ] : [])
+          for_each = var.node_selector != {} ? var.node_selector : {}
```
Comment (lemaitre-aneo) [inline monitoring/onpremise/fluent-bit/main.tf:29]:
```suggestion
          for_each = var.node_selector
```

**Review by lemaitre-aneo — APPROVED** [review]

**Pull request merged**